source/common/quic/envoy_quic_client_connection.cc

```diff
@@ -97,10 +97,19 @@ void EnvoyQuicClientConnection::switchConnectionSocket(
     Network::ConnectionSocketPtr&& connection_socket) {
   auto writer = std::make_unique<EnvoyQuicPacketWriter>(
       std::make_unique<Network::UdpDefaultWriter>(connection_socket->ioHandle()));
+  quic::QuicSocketAddress self_address =
```

Comment:
Should these two addresses only be created in the branch they're used in? `if (connection_migration_use_new_cid())`

Reply:
`connection_socket` will be moved right after this. So we have to latching addresses here.